lib/bolt/executor.rb

```diff
@@ -12,17 +12,19 @@
 require 'bolt/notifier'
 require 'bolt/result_set'
 require 'bolt/puppetdb'
+require 'open3'
 
 module Bolt
   class Executor
     attr_reader :noop, :transports
     attr_accessor :run_as, :plan_logging
 
-    def initialize(config = Bolt::Config.new, analytics = Bolt::Analytics::NoopClient.new, noop = nil)
+    def initialize(config = Bolt::Config.new, analytics = Bolt::Analytics::NoopClient.new, noop = nil, inventory)
```

Comment:
This isn't ideal. It starts to feel weird to have both inventory and executor in Puppet.lookup and also need to have a new class that wraps both that we use for an apply executor.